### DISCUSSION TITLE
AttributeError: 'Session' object has no attribute 'load'

### DISCUSSION THREAD

**Discussion by McHouse** [theOehrly/Fast-F1#312] [Q&A] [answered]
Six months ago I started with this genius of FastF1 and python, and after a few months, I already have several routines that worked perfectly until I upgraded to version 2.3.

even with this code:

session = fastf1.get_session(2022,6,'R')
fastf1.Cache.enable_cache(Route_Cache)
session.load()
session.laps

since then I have been unable to avoid the following error

   File "F:\F1\FastF1-COD\Untitled2.py", line 42, in <module>
     session.load()

AttributeError: 'session' object has no attribute 'load'

Could you please tell me how to load the df laps?
I guess the telemetry and messages have changed too, this: 

session.load(laps=True, telemetry=True, weather=None, messages=True, livedata=None)
laps=session.laps

How do it now?

Thank you

**Comment by McHouse** [comment]
Seeing the requests of other users, I enlarge one of my specific cases, all with the same problem, which I obviously did not have before.


import fastf1 as ff1
import pandas as pd

ff1.Cache.enable_cache("F:/F1/2022/GGPP/CACHE") 
event=ff1.get_event(2022, 7) 
SesionGP=input("Session  (1,2,3,S,Q,R) : ").upper()
if SesionGP=='R':
    Session=event.get_race()
elif SesionGP=='Q':
    Session=event.get_qualifying()
elif SesionGP=='S':
    Session=event.get_sprint()
elif SesionGP=='FP1' or  SesionGP=='1':
    Session=event.get_session('Practice 1')
elif SesionGP=='FP2' or  SesionGP=='2':
    Session=event.get_session('Practice 2')
elif SesionGP=='FP3' or  SesionGP=='3':
    Session=event.get_session('Practice 3')

Session.load(laps=True, telemetry=True, weather=True, messages=True, livedata=None)

laps=Session.laps
Drivers=pd.unique(laps['Driver'])
Ndrivers=int(len(Drivers))
drivers=[0]*Ndrivers
laps_driver=[0]*Ndrivers
fastest_driver=[0]*Ndrivers
telemetry = pd.DataFrame()
Driver_color=[]




---------------------------------------------
and this as result in console:

Python 3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.
IPython 7.29.0 -- An enhanced Interactive Python.
runfile('F:/F1/FastF1-COD/untitled0.py', wdir='F:/F1/FastF1-COD')
Session  (1,2,3,S,Q,R) : r
Traceback (most recent call last):

  File "F:\F1\FastF1-COD\untitled0.py", line 26, in <module>
    Session.load(laps=True, telemetry=True, weather=True, messages=True, livedata=None)

AttributeError: 'Session' object has no attribute 'load'

I hope to help someone who can enlighten me

thank you

**Reply by theOehrly** [reply]
This seems like a really weird problem. Can you please give the output of

```
import fastf1
print(fastf1.__version__)
```

**Comment by McHouse** [comment]
of course 👍 
import fastf1
print(fastf1.__version__)


Python 3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 7.29.0 -- An enhanced Interactive Python.

runfile('C:/Users/McH_JCG/.spyder-py3/temp.py', wdir='C:/Users/McH_JCG/.spyder-py3')
2.3.0

**Comment by McHouse** [comment, marked as answer]
Thanks theOehrly, your indication about what version of Fastf1 I had installed, has led me to uninstall it (pip uninstall fastf1), close Anaconda, reopen the environment and reinstall fastf1 again, this time without added indications like version==2.3, ( pip install fastf1).
After installing successfully, as the process itself indicates, I went back to my routines and... "voilá", they work perfectly (I think, hehehehe).

THANK YOU VERY MUCH, especially for the interest and speed of response. (and because everything has ended well, juas).

**Reply by theOehrly** [reply]
Ok, good that it's fixed. It's a very weird bug because there is no version of FastF1 that has ``fastf1.get_event``  but does not support ``Session.load``. So it really shouldn't be possible that one works and the other doesn't. There also have been no similar reports. I have no idea what the problem was. But as long as this problem doesn't occur more often, it's fine, I guess.
Thank you for reporting back that everything works now.

**Reply by McHouse** [reply]
Thanks for nothing.
I thank you, that your hundreds of hours of keyboarding with this fastf1, we take advantage of all the F1 fans who have not helped you.
THANK YOU for such an excellent job.
